<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SeMainWindow</class>
 <widget class="QMainWindow" name="SeMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1344</width>
    <height>606</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SceneEditor</string>
  </property>
  <property name="windowIcon">
   <iconset resource="images.qrc">
    <normaloff>:/logos/sceneeditor01.png</normaloff>:/logos/sceneeditor01.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="SeMdiArea" name="mdiArea">
    <property name="geometry">
     <rect>
      <x>2</x>
      <y>2</y>
      <width>1073</width>
      <height>549</height>
     </rect>
    </property>
    <property name="viewMode">
     <enum>QMdiArea::SubWindowView</enum>
    </property>
    <widget class="QWidget" name="subwindow">
     <property name="windowTitle">
      <string>Subwindow</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_9">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>1</number>
      </property>
      <property name="topMargin">
       <number>1</number>
      </property>
      <property name="rightMargin">
       <number>1</number>
      </property>
      <property name="bottomMargin">
       <number>1</number>
      </property>
      <item>
       <widget class="SeSceneView" name="gview"/>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1344</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExi"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout_Qt"/>
    <addaction name="actionAbout_SceneEditor"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="allowedAreas">
    <set>Qt::LeftToolBarArea|Qt::TopToolBarArea</set>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonIconOnly</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionLoad"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionSave_As"/>
   <addaction name="separator"/>
   <addaction name="actionNew"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="dockScenes">
   <property name="minimumSize">
    <size>
     <width>374</width>
     <height>462</height>
    </size>
   </property>
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Scenes</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_7">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>1</number>
     </property>
     <property name="topMargin">
      <number>1</number>
     </property>
     <property name="rightMargin">
      <number>1</number>
     </property>
     <property name="bottomMargin">
      <number>1</number>
     </property>
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="styleSheet">
        <string notr="true">font: 10pt &quot;MS Shell Dlg 2&quot;;</string>
       </property>
       <property name="tabPosition">
        <enum>QTabWidget::East</enum>
       </property>
       <property name="tabShape">
        <enum>QTabWidget::Triangular</enum>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tab">
        <attribute name="title">
         <string>Layer</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <property name="spacing">
          <number>5</number>
         </property>
         <property name="leftMargin">
          <number>1</number>
         </property>
         <property name="topMargin">
          <number>1</number>
         </property>
         <property name="rightMargin">
          <number>1</number>
         </property>
         <property name="bottomMargin">
          <number>1</number>
         </property>
         <item>
          <widget class="SeTreeScenes" name="treeScenes">
           <attribute name="headerVisible">
            <bool>false</bool>
           </attribute>
           <column>
            <property name="text">
             <string notr="true">1</string>
            </property>
           </column>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBoxMosaic">
           <property name="title">
            <string>Mosaic and Transition</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item>
               <widget class="QLabel" name="label_5">
                <property name="text">
                 <string>Image:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="lblImageInfo">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>-.-</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="cmdSelectImage">
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="images.qrc">
                  <normaloff>:/Images/img.png</normaloff>:/Images/img.png</iconset>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="cmdModifyReset">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="images.qrc">
                  <normaloff>:/Images/clear.png</normaloff>:/Images/clear.png</iconset>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="cmdModifyImage">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="images.qrc">
                  <normaloff>:/Images/modify.png</normaloff>:/Images/modify.png</iconset>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="QRadioButton" name="radioShowRectangle">
                <property name="text">
                 <string>Rectangle</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="radioShowCircles">
                <property name="text">
                 <string>Circles</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QCheckBox" name="chkLayerEnabled">
                <property name="text">
                 <string>Enabled</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QLabel" name="label_6">
                <property name="text">
                 <string>Delay:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="spinDelay">
                <property name="decimals">
                 <number>3</number>
                </property>
                <property name="maximum">
                 <double>50.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.250000000000000</double>
                </property>
                <property name="value">
                 <double>1.000000000000000</double>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="chkLoop">
                <property name="text">
                 <string>Loop</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBoxPlay">
           <property name="title">
            <string/>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QPushButton" name="cmdDeploy">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="toolTip">
               <string>Create deployment files, i.e. a simple CSV file with a set of color codes.</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="images.qrc">
                <normaloff>:/Images/deploy.png</normaloff>:/Images/deploy.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="cmdGenerateVideo">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="toolTip">
               <string>Generate a video based on the sequences and open destination folder.</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="images.qrc">
                <normaloff>:/Images/video.png</normaloff>:/Images/video.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="cmdDeployWebSocket">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="toolTip">
               <string>Open a WebSocket connection to a RGB wall and deploy the current shown layer.</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="images.qrc">
                <normaloff>:/Images/websocket0.png</normaloff>:/Images/websocket0.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="lblDeployWebSocket">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_4">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="cmdStartAnimation">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Start</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="cmdStopAnimation">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Stop</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab_2">
        <attribute name="title">
         <string>Properties</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <property name="spacing">
          <number>5</number>
         </property>
         <property name="leftMargin">
          <number>1</number>
         </property>
         <property name="topMargin">
          <number>1</number>
         </property>
         <property name="rightMargin">
          <number>1</number>
         </property>
         <property name="bottomMargin">
          <number>1</number>
         </property>
         <item>
          <widget class="QGroupBox" name="groupBoxProperties">
           <property name="title">
            <string>Properties</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <property name="spacing">
             <number>5</number>
            </property>
            <property name="leftMargin">
             <number>5</number>
            </property>
            <property name="topMargin">
             <number>5</number>
            </property>
            <property name="rightMargin">
             <number>5</number>
            </property>
            <property name="bottomMargin">
             <number>5</number>
            </property>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_2">
                <item>
                 <widget class="QLabel" name="label">
                  <property name="text">
                   <string>Column:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="label_2">
                  <property name="text">
                   <string>Row:</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout">
                <item>
                 <widget class="QLineEdit" name="linePropertyColumn"/>
                </item>
                <item>
                 <widget class="QLineEdit" name="linePropertyRow"/>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>Change mode:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="cmbChangeMode">
                <item>
                 <property name="text">
                  <string extracomment="The color between two layer will be changed immediatelly.">Hard</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string extracomment="Between two layer the color will be faded to the new color.">Fade</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QLabel" name="label_4">
                <property name="text">
                 <string>Color:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="cmdChangeColor">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="styleSheet">
                 <string notr="true">border-color: rgb(0, 0, 0);</string>
                </property>
                <property name="text">
                 <string/>
                </property>
                <property name="flat">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>383</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBoxMultiselection">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="title">
            <string>Properties (Multiselection)</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_8">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>Change mode:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="cmbChangeMode_2">
                <item>
                 <property name="text">
                  <string notr="true">--</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string extracomment="The color between two layer will be changed immediatelly.">Hard</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string extracomment="Between two layer the color will be faded to the new color.">Fade</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_10">
              <item>
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>Color:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="cmdChangeColor_2">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="styleSheet">
                 <string notr="true">border-color: rgb(0, 0, 0);</string>
                </property>
                <property name="text">
                 <string/>
                </property>
                <property name="flat">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_11">
              <item>
               <widget class="QPushButton" name="cmdUnselect">
                <property name="text">
                 <string>Unselect</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_3">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QPushButton" name="cmdSelectAll">
                <property name="text">
                 <string>Select All</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionLoad">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/Images/open.png</normaloff>:/Images/open.png</iconset>
   </property>
   <property name="text">
    <string>Load</string>
   </property>
   <property name="toolTip">
    <string>Load a previous scene configuration.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/Images/save.png</normaloff>:/Images/save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Saves the current scene configuration.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/Images/saveas.png</normaloff>:/Images/saveas.png</iconset>
   </property>
   <property name="text">
    <string>Save As...</string>
   </property>
   <property name="toolTip">
    <string>Saves the current scene configuration.</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/Images/new.png</normaloff>:/Images/new.png</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
   <property name="toolTip">
    <string>Clear current scene and start creating a new one.</string>
   </property>
  </action>
  <action name="actionExi">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/Images/exit.png</normaloff>:/Images/exit.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="text">
    <string>About Qt...</string>
   </property>
  </action>
  <action name="actionAbout_SceneEditor">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/logos/sceneeditor01.png</normaloff>:/logos/sceneeditor01.png</iconset>
   </property>
   <property name="text">
    <string>About SceneEditor...</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>SeTreeScenes</class>
   <extends>QTreeWidget</extends>
   <header location="global">SeTreeScenes.h</header>
  </customwidget>
  <customwidget>
   <class>SeMdiArea</class>
   <extends>QMdiArea</extends>
   <header location="global">SeMdiArea.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SeSceneView</class>
   <extends>QGraphicsView</extends>
   <header location="global">SeSceneView.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="images.qrc"/>
 </resources>
 <connections/>
</ui>
